lib/streamlit/file_util.py

```diff
@@ -137,3 +137,34 @@ def file_is_in_folder_glob(filepath, folderpath_glob):
 
     file_dir = os.path.dirname(filepath) + "/"
     return fnmatch.fnmatch(file_dir, folderpath_glob)
+
+
+def file_in_pythonpath(filepath):
+    """Test whether a filepath is in the same folder of a path specified in the PYTHONPATH env variable.
+
+
+    Parameters
+    ----------
+    filepath : str
+        An absolute file path.
+
+    Returns
+    -------
+    boolean
+        True if contained in PYTHONPATH, False otherwise. False if PYTHONPATH is not defined or empty.
+
+    """
+
+    if "PYTHONPATH" not in os.environ:
+        return False
+    pythonpath = os.environ["PYTHONPATH"]
+    if len(pythonpath) == 0:
+        return False
+
+    absolute_paths = [
+        os.path.abspath(path) for path in pythonpath.split(os.pathsep)
+    ]
```

Comment:
(Just a question) - how expensive is this? Are we shooting ourselves in the foot, performance-wise, by not caching the `absolute_paths` value? (Obviously this isn't appropriate if PYTHONPATH can change during execution; I'm assuming it can't but maybe I'm wrong.)

Reply:
You mean because I am applying `abspath` to some absolute paths? `PYTHONPATH` should not be very big, so I would say this is not a bottleneck. The alternative is checking with `isabs`, but not sure it is worth it.

Reply:
I guess... you mean by caching... yeah, given that this can change I would not do it right now... let me think about performance a bit here...

Reply:
In general, if the performance optimization can introduce a correctness issue, I would opt out unless we think it is going to be a bottleneck.

Reply:
Won't do the cache, but I left the check for the pythonpath as the last one in the boolean checks below.